torchaudio/functional.py

```diff
@@ -310,11 +310,15 @@ def griffinlim(
         rand_init (bool): Initializes phase randomly if True, to zero otherwise. (Default: ``True``)
 
     Returns:
-        torch.Tensor: waveform of (channel, time), where time equals the ``length`` parameter if given.
+        torch.Tensor: waveform of (..., time), where time equals the ``length`` parameter if given.
     """
     assert momentum < 1, 'momentum=%s > 1 can be unstable' % momentum
     assert momentum > 0, 'momentum=%s < 0' % momentum
 
+    # pack batch
+    shape = specgram.size()
+    specgram = specgram.reshape([-1] + list(shape[-2:]))
```

Comment:
Instead of reshape I suggest you use view. Reshape dispatches to view if it doesn't require a copy. If this does require a copy I think it'd be good to know about since it's not something we'd expect to be necessary.

Reply:
Good idea, but that means I need to change all of the batching. I suggest to change that all at the same time explicitly as a separate PR.